### PR TITLE
Pulsar Component improvements (Auth, Metadata, Logging)

### PR DESCRIPTION
**Pull request by crtomirmajer**
Addressing proposals in #917

### CODE REVIEW

File: internal/impl/pulsar/input.go [hunk 1]
```diff
@@ -38,8 +44,14 @@ func init() {
 This input adds the following metadata fields to each message:
 
 ` + "```text" + `
+- pulsar_message_id
 - pulsar_key
+- pulsar_ordering_key
+- pulsar_event_time
+- pulsar_publish_time
```
[the remark this answers is not in the document]
Comment (Jeffail) [reply internal/impl/pulsar/input.go:51]: Hey, sorry I missed this. I think I'd prefer `pulsar_event_time_unix` for consistency. Everything else looks great.

**Comment by crtomirmajer** [comment]
@Jeffail, @mihaitodor, could you please approve running the workflow, and take another look at the PR?

**Comment by Jeffail** [comment]
Hey @crtomirmajer, PR looks great, is this functionality blocking you? I'm happy to merge if the tests pass but ideally I want to get v3.59.0 out the door first, if you need this ASAP I'll squeeze it in.

**Comment by crtomirmajer** [comment]
@Jeffail, it can wait.

**Comment by Jeffail** [comment]
Okay we're good to go, thanks @crtomirmajer!

**Pull request merged**